src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs

```diff
@@ -39,7 +39,8 @@ public class TestRequestHandler : ITestRequestHandler
 
         private ManualResetEventSlim sessionCompleted;
 
-        private Action<Message> onAckMessageRecieved;
+        private Action<Message> onLaunchAdapterProcessWithDebuggerAttachedAckReceived;
+        private Action<Message> onAttachDebuggerAckRecieved;
```

Comment:
After addressing part of the code review comments I changed the lock in `DesignModeClient.cs` to avoid it being message specific.
It was like that in the first place because I thought that we might get two simultaneous callbacks, however after some research it became clear to me that the two scenarios were mutually exclusive.